lib/krane/cluster_resource_discovery.rb

```diff
@@ -17,13 +17,38 @@ def crds
     end
 
     def global_resource_kinds
-      @globals ||= fetch_globals.map { |g| g["kind"] }
+      @globals ||= fetch_resources(only_globals: true).map { |g| g["kind"] }
+    end
+
+    def prunable_resources
+      api_versions = fetch_api_versions
+      fetch_resources.map do |resource|
+        next unless resource['verbs'].include?("delete")
+        version = api_versions.fetch(resource['apigroup'], ['v1']).last
```

Comment:
I'm a bit confused by this--it seems to imply that the api versions list can be missing a resource's API group. Isn't the logic actually "if the resource has no api group, then the version is v1; otherwise, grab it from the api versions list"? 

```ruby
version = resource["apigroup"].blank? ? 'v1' : api_versions["apigroup"].last
```

Also, why is the last one the correct choice? They seem to be alphabetical, so this would make us pick beta over alpha, but also beta over GA. Does it matter?

Reply:
> Also, why is the last one the correct choice? They seem to be alphabetical, so this would make us pick beta over alpha, but also beta over GA. Does it matter?

We get lucky and for global, last just happens to work. The PR to add this to namespaced resources actually parses the versions, pickets the latest and has overides for a resource isn't supported in the latest version of the group.
https://github.com/Shopify/kubernetes-deploy/pull/616/files#diff-3c59dc6581aa611cc9f2c1c012d0f539R28

Reply:
I think the answer is yes, but to confirm: if we pick a version that doesn't work, do we 100% for sure find out in our CI, even for resources that aren't in our test suite? (otherwise should we be making an API call during the deploy to validate our choice?)

Reply:
The answer is yes. `kubectl apply` fails so we get lots of failures for global-deploy and deploy.